src/cascadia/TerminalApp/SettingsTab.cpp

```diff
@@ -0,0 +1,177 @@
+
+            // Update SwitchToTab command's icon
+            SwitchToTabCommand().IconSource(GetFontIcon<winrt::WUX::Controls::IconSource>(fontFamily, 12, glyph));
+        }
+    }
+
+    // Method Description:
+    // - Gets the title string of the settings UI
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - the title string of the last focused terminal control in our tree.
+    winrt::hstring SettingsTab::GetActiveTitle() const
+    {
+        // TODO: This _could_ change depending on what page of the settings UI is open/focused?
+        return Title();
+    }
+
+    // Method Description:
+    // - Prepares this tab for being removed from the UI hierarchy
+    void SettingsTab::Shutdown()
+    {
+        // TODO: Does/Will the settings UI need some shutdown procedures?
+        Content(nullptr);
+        _ClosedHandlers(nullptr, nullptr);
+    }
+
+    // Method Description:
+    // - Creates a context menu attached to the tab.
+    // Currently contains elements allowing the user to close the selected tab
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void SettingsTab::_CreateContextMenu()
+    {
+        auto weakThis{ get_weak() };
+
+        // Close
+        Controls::MenuFlyoutItem closeTabMenuItem;
+        Controls::FontIcon closeSymbol;
+        closeSymbol.FontFamily(Media::FontFamily{ L"Segoe MDL2 Assets" });
+        closeSymbol.Glyph(L"\xE8BB");
+
+        closeTabMenuItem.Click([weakThis](auto&&, auto&&) {
+            if (auto tab{ weakThis.get() })
+            {
+                tab->_ClosedHandlers(nullptr, nullptr);
+            }
+        });
+        closeTabMenuItem.Text(RS_(L"TabClose"));
+        closeTabMenuItem.Icon(closeSymbol);
+
+        // Build the menu
+        Controls::MenuFlyout newTabFlyout;
+        Controls::MenuFlyoutSeparator menuSeparator;
+        newTabFlyout.Items().Append(closeTabMenuItem);
+        TabViewItem().ContextFlyout(newTabFlyout);
+    }
+
+    // Method Description:
+    // - Initializes a SwitchToTab command object for this Tab instance.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void SettingsTab::_MakeSwitchToTabCommand()
+    {
+        auto focusTabAction = winrt::make_self<implementation::ActionAndArgs>();
+        auto args = winrt::make_self<implementation::SwitchToTabArgs>();
+        args->TabIndex(_TabViewIndex);
+
+        focusTabAction->Action(ShortcutAction::SwitchToTab);
+        focusTabAction->Args(*args);
+
+        winrt::TerminalApp::Command command;
+        command.Action(*focusTabAction);
+        command.Name(Title());
+        command.IconSource(IconSource());
+
+        SwitchToTabCommand(command);
+    }
+
+    void SettingsTab::UpdateTabViewIndex(const uint32_t idx)
+    {
+        TabViewIndex(idx);
+        SwitchToTabCommand().Action().Args().as<implementation::SwitchToTabArgs>()->TabIndex(idx);
+    }

```

Comment:
These are two other functions that are probably the same across both `ITab` implementations. Could we consolidate them somehow?

Reply:
Hmm, `_MakeSwitchToTabCommand` I can put inside of `TerminalPage`, the other function should actually be modified so that the setter for `TabViewIndex` also sets `SwitchToTabCommand`'s tabIndex.